dgisim/src/dices.py

```diff
@@ -386,49 +384,30 @@ def comparision(x: tuple[Element, int]) -> tuple[int, int, int]:
             ]
 
             least_spend = min(
-                elements_how_many_omni_to_spent_if_fill_omni,
+                elem_cost_mapping,
                 key=comparision
             )[1]  # type: ignore
             least_spend_omni_elements: list[Element] = []
-            if least_spend != BIG_INT:
-                for elmnt, omni_to_spent in elements_how_many_omni_to_spent_if_fill_omni:
-                    if omni_to_spent == least_spend:
-                        least_spend_omni_elements.append(elmnt)
-            else:
-                # Cant feed OMNI requirement
+            if least_spend == BIG_INT:
+                # Cant fill OMNI requirement
                 return None
 
-            assert least_spend_omni_elements, "Unknown error"
+            for elem, omni_to_spent in elem_cost_mapping:
+                if omni_to_spent == least_spend:
+                    least_spend_omni_elements.append(elem)
```

Comment:
Could you make this part a list-comprehension (for `least_spent_omni_elems`)

Reply:
done